src/controllers/userController.js

```diff
@@ -0,0 +1,81 @@
+const jwt = require("jsonwebtoken")
+const { checkEmptyValues, generateToken, throwError } = require("../utils");
+const { userServicve } = require("../services");
+const { detailUpdateUser } = userServicve;
+
+const detailSignUp = async (req, res) => {
+
+  const accessToken = req.headers.authorization;
+  const { id } = jwt.verify(accessToken, process.env.SECRET);
+  const userId = id;
+
+  if(!accessToken) throwError("INVALID_TOKEN")
+  
+  const {
+    userType,
+    nickname,
+    phoneNumber,
+    gender,
+    birthday,
+    height,
+    weight,
+    interestedWorkout,
+    workoutLoad,
+    specialized,
+  } = req.body;
+  
+  try {
+    // 키에러 체크
+    if (!userType) throwError(400, "KEY_ERROR");
+
+    const commonFields = [
+      nickname,
+      phoneNumber,
+      gender,
+      birthday,
+      height,
+      weight,
+      interestedWorkout,
+      workoutLoad,
+    ];
+
+    //트레이너만 specialized 입력
+    if (userType === "1" || userType === "2") {
+      const fieldsToCheck = userType === "2" ? [...commonFields, specialized] : commonFields;
+
+      checkEmptyValues(...fieldsToCheck);
+    } else {
+      throwError(400, "INVALID_USER_TYPE");
+    }
```

Comment:
```suggestion
    ];

    //트레이너만 specialized 입력
    const userTypes = {
      USER : 1,
      TRAINER : 2
    }
    if (userType === userTypes.USER || userType === userTypes.TRAINER) {
      const fieldsToCheck = userType === userTypes.TRAINER ? [...commonFields, specialized] : commonFields;

      checkEmptyValues(...fieldsToCheck);
    } else {
      throwError(400, "INVALID_USER_TYPE");
    }
```

1. 1번과 2번이 어떤 타입인지 개발자가 모두 외우고 있을 수 없기 때문에, 위와 같이 타입을 객체로 명시하는 방법을 자주 사용합니다.